### ISSUE TITLE
Web component: Link List - Beginning and Ending boundary for the cards

### ISSUE THREAD

**Issue by Praveen-sr** [carbon-design-system/carbon-for-ibm-dotcom#4060]
## Detailed description
On  JAWS FF & VO iPhone safari

Issue: The beginning and ending boundary for each card is not conveyed for the screen reader user. 
Note: Working fine in React Staging

Expected : Provide each card within the landmark region and give appropriate label.

**Comment by asudoh** [comment]
Here's what I see in React storybook, not sure if the behavior, especially with the wider width, is something we should mirror to Web Components. Any thoughts...? Thanks! CC @oliviaflory @JennySanchez @RaphaelAmadeu 

![linklist-responsive-padding](https://user-images.githubusercontent.com/1259051/94703767-2b2a0680-037a-11eb-891a-f15eada7ac9a.gif)

**Comment by ariellalgilmore** [comment]
@Praveen-sr is it possible to get more info on why this should be labeled a `region`. The link list cards are passed in `href` values meaning this is a `link` item. Should the possible fix instead be changing the React version to a `listitem` role instead of a `region` role

cc: @asudoh
